### ISSUE TITLE
CV and training times - exponential growth 

### ISSUE THREAD

**Issue by deakkon** [keras-team/keras-tuner#338]
Hi all,

great work and thank you for releasing this to the public. It elevates a lot of issues with manual fiddling with the model. 

I have an issue with training times when using it in a for loop, over CV folds. The training times over the first folds are approx. 1 hour (on a CPU; the times differ but the behavior I describe next is the same on a GPU). 

When it comes to the second (and any subsequent) fold, the training times increase dramatically. Its been almost an hour since the training on the 2nd fold was started and I still do not have the first trial summary. It does what it does eventually, but as you can imagine training over e.g. 10 folds is rather lengthy. Especially as I do not see any logical reason fur such behavior. I feel this has more to do with the for loop than what I iterate over? 

I have tried to debug it myself but alas to no avail. Could you suggest anything I could try/look at to mitigate this? It is odd behavior. I am usingTF2.1 (via tf.keras) and the latest version of the tuner. 

Any suggestions would be appreciated. 

Here are the relevant snippets in my code:

        for split_index, split in self.get_splits(splits):
            logger.info("Training on split {}/{}".format(split_index + 1, len(splits)))
            neural_model = self.build_neural_model()

            # TRAIN DATA WRANGLING

            #    TRAIN MODEL     #
            tuner = RandomSearch(
                hypermodel=neural_model.get_model,
                objective=kerastuner.Objective("f1", direction="max"),
                max_trials=self.training_parameters.max_trails,
                executions_per_trial=self.training_parameters.executions_per_trail,
                directory="reports/profiling/{}/{}/{}".format(
                    self.training_parameters.model_name,
                    self.get_date_string(),
                    rand_uuid
                ),
                project_name="split_{}".format(split_index),
                overwrite=True,
                seed=42
            )

            tuner.search(
                train_data,
                train_targets,
                batch_size=self.training_parameters.batch_size,
                epochs=self.training_parameters.epochs,
                verbose=0,
                callbacks=neural_model.get_model_callbacks(
                    model_name=self.training_parameters.model_name,
                    optimization_value="f1",
                    validation_data=(
                        test_data,
                        test_targets,
                    ),
                    digits=None,
                    verbose=0
                ),
                shuffle=True
            )
            # tuner.results_summary()
            best_model = tuner.get_best_models()[0]

**Comment by deakkon** [comment]
Some approx. two hours later, the summary of the first trial was printed to the screen:

#[Trial summary]
 |-Trial ID: 40754ed9ef00f75963e268efcb76e30b
 |-Score: 0.9976956629084026
 |-Best step: 0
 > Hyperparameters:
 |-concat_bilstm_units: 128
 |-concat_rec_drop: 0.1
 |-dropout_dense: 0.5
 |-positional_embeddings: 100
 |-td_char_enc_dropout: 0.1
 |-td_char_enc_recurrent_dropout: 0.1
 |-td_dense_units: 96

What is alarming here is the score; on the first fold, the average trail, across all trials, the score was roughly 0.5 while all experiments in the second fold, for some reason, have a score of almost 1 (which is obviously flawed). 

Any ideas what is causing this discrepancy in the score value or where should I shart to look?

**Comment by deakkon** [comment]
Silly mistake on my part! My appologies!
